### ISSUE TITLE
Window frame size changed callback

### ISSUE THREAD

**Issue by ghuser404** [glfw/glfw#1509]
I know this probably happens approximately once in a decade, and I'm not sure how difficult it is to add support for that, but I think it should be there - even if just for completeness.

**Comment by elmindreda** [comment]
That's true and with Windows non-client scaling it can happen fairly frequently, but it's probably rare that people need to know about it?

Since GLFW is turning into more of a generic top-level window library rather than just _stuff a game needs_ this makes sense to add, at the very least for 4.0 when events should hopefully be easier to add.
